src/iso/CMakeLists.txt

```diff
@@ -19,4 +19,5 @@ add_library(iso STATIC
 
 target_link_libraries(iso
   fmt
+  utils
```

Comment:
Indeed, smaller entity of the utils and the client library only links to what is needed would be better. However, do we want to make a separate library for that or just moving them to a separate file and keep the whole utils lib? Besides that, we may also want to apply the same pattern to other "sub-utils". I think we can discuss this in our sync.

Reply:
I think I can still make that set of utils a different cpp file, it will stay in the same utils lib, but at least we have the header inclusion finer grained granularity.

Reply:
Sounds good!